tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use virt_autotest::common;
+use virt_autotest::utils qw(is_xen_host start_guests check_port_state check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my %health_status = ();
+    @health_status{(keys %virt_autotest::common::guests)} = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (keys %health_status) {
+        script_run("virsh start $guest");
+        if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 2, retry => 60, die => 0) == 0) {
+            $health_status{$guest} = check_guest_health($guest);
+        }
+        else {
+            $health_status{$guest} = "Unreachable";
+        }
+        record_info("$guest: $health_status{$guest}", result => 'fail') unless $health_status{$guest} eq 'pass';
+    }
+    my @unhealthy_list = grep { $health_status{$_} ne 'pass' } keys %health_status;
+
+    script_run("tar libvirtd.tar.gz /var/log/libvirt", die_on_timeout => 0);
+    script_run("tar xen_logs.tar.gz /var/log/xen", die_on_timeout => 0) if is_xen_host;
+    my @host_extra_logs = qw(libvirtd.tar.gz xen_logs.tar.gz);
+    if (get_var('FORCE_UPLOAD_LOGS')) {
+        collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status), @host_extra_logs);
+    }
+    else {
+        collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } @unhealthy_list)) unless @unhealthy_list == 0;
+    }
```

Comment:
You need to put this block
```
    if (get_var('FORCE_UPLOAD_LOGS')) {
        collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status), @host_extra_logs);
    }
    else {
        collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } @unhealthy_list)) unless @unhealthy_list == 0;
    }

```
into post_fail_hook.  There is no point to force upload all logs if all is healthy. This dose not make sense and very costly. IMHO.

Reply:
No, I don't think so. I was requested to upload logs for a 'working' case in bsc#1204882 in comment 4 and comment 8. I had to manually reproduce and collected all logs. I am tired of this.

Even with FORCE_UPLOAD_LOGS, all logs will be uploaded. We don't set it by default in test suites, we only use it when we need it.